xds/src/main/java/io/grpc/xds/internal/security/ClientSslContextProviderFactory.java

```diff
@@ -48,8 +48,9 @@ public SslContextProvider create(UpstreamTlsContext upstreamTlsContext) {
     checkNotNull(
         upstreamTlsContext.getCommonTlsContext(),
         "upstreamTlsContext should have CommonTlsContext");
-    if (CommonTlsContextUtil.hasCertProviderInstance(
-        upstreamTlsContext.getCommonTlsContext())) {
+    if (CommonTlsContextUtil.hasCertProviderInstance(upstreamTlsContext.getCommonTlsContext())
+        || upstreamTlsContext.getCommonTlsContext().getCombinedValidationContextOrBuilder()
+            .getDefaultValidationContext().hasSystemRootCerts()) {
```

Comment:
There should be something somewhere that is protected by GRPC_EXPERIMENTAL_XDS_SYSTEM_ROOT_CERTS. It looks like this change isn't yet supporting system_root_certs because it didn't change the validation logic in XdsClusterResource?

It is so very, very bad that the prexisting code is passing Downstream/UpstreamTlsContext this deep, and re-performing validation at this point.

Reply:
Added validation logic in XdsClusterResource.